src/blue/BlueSnippets.sol

```diff
@@ -273,6 +273,32 @@ contract BlueSnippets {
         (assetsWithdrawn, sharesWithdrawn) = morpho.withdraw(marketParams, amount, supplyShares, onBehalf, receiver);
     }
 
+    /**
+     * @notice Handles the withdrawal of a specified amount of assets by a user from a specific market. If the amount is
+     * greater than the total amount suplied by the user, withdraws all the shares of the user.
+     * @param marketParams The parameters of the market.
+     * @param amount The amount of assets the user is withdrawing.
+     * @param user The address of the user withdrawing the assets.
+     * @return assetsWithdrawn The actual amount of assets withdrawn.
+     * @return sharesWithdrawn The shares withdrawn in return for the assets.
+     */
+    function withdrawAmountOrAll(MarketParams memory marketParams, address user, uint256 amount)
+        external
+        returns (uint256 assetsWithdrawn, uint256 sharesWithdrawn)
+    {
+        uint256 shares;
+        address onBehalf = user;
+        address receiver = user;
+        uint256 totalSupplyAssets = morpho.expectedSupplyAssets(marketParams, user);
```

Comment:
we can accrue interest to save gas

Reply:
with a comment indicating that if `view` is required (like in a `getWithdrawAmount` function), one can use `MorphoBalancesLib.expectedSupplyAssets`